pkg/test/framework/components/prometheus/kube.go

```diff
@@ -181,28 +181,34 @@ func (c *kubeComponent) WaitForOneOrMore(format string, args ...interface{}) err
 
 		switch v.Type() {
 		case model.ValScalar, model.ValString:
-			return nil, true, nil
+			return v, true, nil
 
 		case model.ValVector:
 			value := v.(model.Vector)
 			value = reduce(value, map[string]string{})
 			if len(value) == 0 {
 				return nil, false, fmt.Errorf("value not found (query: %q)", query)
 			}
-			return nil, true, nil
+			return v, true, nil
 
 		default:
 			return nil, true, fmt.Errorf("unhandled value type: %v", v.Type())
 		}
 	}, retryTimeout, retryDelay)
 
-	return err
+	var v model.Value
+	if value != nil {
```

Comment:
is the assumption here that if err == nil, value will always be a valid `model.Value`?